lms/djangoapps/courseware/block_render.py

```diff
@@ -548,14 +548,10 @@ def inner_get_block(block):
         if staff_access:
             block_wrappers.append(partial(add_staff_markup, user, disable_staff_debug_info))
 
-    field_data = DateLookupFieldData(block._field_data, course_id, user)  # pylint: disable=protected-access
-    field_data = LmsFieldData(field_data, student_data)
-
     store = modulestore()
 
     services = {
         'fs': FSService(),
-        'field-data': field_data,
```

Comment:
I did not find any direct usage of this service.

We bind the field_data in the `XModuleMixin`: https://github.com/openedx/edx-platform/blob/0b455e0336fa54199fd992c2ce83a81c07dd07c2/xmodule/x_module.py#L649-L659

Then, we have the following wrappers for accessing `field-data`; none of them uses the service.

Mongo: https://github.com/openedx/edx-platform/blob/dac1c5abe1193ce51297d6aa6db32742b2f6cedc/xmodule/modulestore/mongo/base.py#L317-L322

Split Mongo: https://github.com/openedx/edx-platform/blob/fbaa2e5a68374502d10de95faf60cdb009f83715/xmodule/modulestore/split_mongo/caching_descriptor_system.py#L226-L239

New runtime: https://github.com/openedx/edx-platform/blob/aa85257b1913b3837216f2c3412e5c1e7114398e/openedx/core/djangoapps/xblock/runtime/runtime.py#L227-L235

Reply:
@Agrendalath, since the service is being removed, do we need [@XBlock.needs("field-data")](https://github.com/openedx/edx-platform/blob/3fc1f92118315d61f74f2a84a852d65d44cbca8b/cms/djangoapps/contentstore/views/tests/test_preview.py#L175C28-L175C28)?

Reply:
Regarding removing `@XBlock.needs('field-data')`: The `field-data` service in general is still used; it's just this particular way of setting the student-bound wrapped field data that is no longer used. I think :) 

Though I also think that the field-data service is always "needed" so it's probably not necessary to use the @XBlock.needs decorator.

Reply:
Nice catch! We use `field-data` as a service but never reach the service declaration checks because of its special handling. I removed these decorators in cca07d6.